src/include/duckdb/execution/operator/join/physical_hash_join.hpp

```diff
@@ -56,12 +63,16 @@ class PhysicalHashJoin : public PhysicalComparisonJoin {
 public:
 	// Source interface
 	unique_ptr<GlobalSourceState> GetGlobalSourceState(ClientContext &context) const override;
+	unique_ptr<LocalSourceState> GetLocalSourceState(ExecutionContext &context,
+	                                                 GlobalSourceState &gstate) const override;
 	void GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate,
 	             LocalSourceState &lstate) const override;
 
+	//! Becomes a source when it is an external join
```

Comment:
How so? Right now the control flow of the operator is:
1. Sink operator interface for build side
2. Operator interface for probing
3. Source operator interface for partitioned probing (anything that we couldn't probe in (2.))

It has to always be marked as a source because we need to schedule the source pipeline before execution (it can decide at any point during execution to go external and become a source op).

Reply:
Sorry not sure what I was thinking there - it makes sense now.